glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -77,6 +80,12 @@ open class GleanInternalAPI internal constructor () {
         this.gleanDataDir = File(applicationContext.applicationInfo.dataDir, "glean_data")
         handle = LibGleanFFI.INSTANCE.glean_initialize(this.gleanDataDir.path, applicationContext.packageName)
 
+        // If any pings were registered before initializing, do so now
+        this.pingTypeQueue.forEach { this.registerPingType(it) }
+        if (!Dispatchers.API.testingMode) {
```

Comment:
Why do we keep this queue around in testing mode?

Reply:
It's, uh, complicated...

PingTypes are registered by virtue of their being created.  They are singletons.  Therefore, they only get registered once (regardless of when `Glean.registerPings()` is called -- it only works the first time).  This is because we don't want to use introspection and don't want to have an explicit data structure listing all the available pings.

So... in testing, we want to "rerun" the queue every time.  Outside of testing, this is just wasted memory.

But I'd be happy to just delete this clause altogether.  It's not a lot of wasted memory.